packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -245,9 +323,19 @@ export function navigateReducer(
     )
   }
 
+  state.prefetchCache.set(createHrefFromUrl(url, false), {
+    data: Promise.resolve(cache.data!),
+    // this will make sure that the entry will be discarded after 30s
+    kind: 'temporary',
+    prefetchTime: Date.now(),
+    treeAtTimeOfPrefetch: state.tree,
+    lastUsedTime: Date.now(),
+  })
+
   // Unwrap cache data with `use` to suspend here (in the reducer) until the fetch resolves.
   const [flightData, canonicalUrlOverride] = readRecordValue(cache.data!)
 
+  console.log('reading flight data', flightData)
```

Comment:
q for Seb/TIm: we can technically remove anything below that point because the ^ suspend just above will force the reducer to re-run and then it will run as-if it was a regular prefetch/navigation now. This is because we store every navigation as a temporary prefetch node now.

the only case where it would not is where you'd have a `cache.data` already resolved I guess but I'm not sure where that would happen?